include/zephyr/kernel/internal/sched_priq.h

```diff
@@ -44,15 +48,15 @@ void z_priq_rb_remove(struct _priq_rb *pq, struct k_thread *thread);
 struct k_thread *z_priq_rb_best(struct _priq_rb *pq);
 
 /* Traditional/textbook "multi-queue" structure.  Separate lists for a
- * small number (max 32 here) of fixed priorities.  This corresponds
+ * small number of fixed priorities.  This corresponds
  * to the original Zephyr scheduler.  RAM requirements are
  * comparatively high, but performance is very fast.  Won't work with
  * features like deadline scheduling which need large priority spaces
  * to represent their requirements.
  */
 struct _priq_mq {
-	sys_dlist_t queues[32];
-	unsigned int bitmask; /* bit 1<<i set if queues[i] is non-empty */
+	sys_dlist_t queues[PRIQ_BITMAP_SIZE * NBITS];
```

Comment:
Question -- could we not use `sys_dlist_t queues[NUM_PRIO];` ? (Both z_setup_new_thread() and k_thread_priority_set() have a Z_ASSERT_VALID_PRIO check to verify that the priority is in range so we should not be accessing beyond the bounds of the array.)

Reply:
Yes, if the condition already is checked.